### PR TITLE
Move activeness check of globalKeyInputs into archive

### PR DESCRIPTION
**Pull request by cocreature**
This is semantically equivalent but it seems slightly simpler and more
importantly it leads to slightly nicer semantics by reducing the
number of cases where we have to worry about accidentally skipping an
activeness check.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala [hunk 1]
```diff
@@ -1299,12 +1299,8 @@ private[lf] object SBuiltin {
           // Check if we have a cached global key result.
           onLedger.ptx.globalKeyInputs.get(gkey) match {
             case Some(optCid) =>
```
Comment (remyhaemmerle-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala:1301]:
```suggestion
            case Some(keyMapping) =>
```

Comment (cocreature) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala]: https://github.com/digital-asset/daml/pull/9610/commits/91d237cf8337dbc2f8dd7812aa4edf5d630ce358

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala [hunk 2]
```diff
@@ -486,10 +486,13 @@ private[lf] case class PartialTransaction(
...
                 val gkey = GlobalKey(templateId, kWithM.key)
-                keys.get(gkey) match {
+                (keys.get(gkey), globalKeyInputs.get(gkey)) match {
                   // An archive can only mark a key as inactive
                   // if it was brought into scope before.
-                  case Some(KeyActive(cid)) if cid == targetId => keys.updated(gkey, KeyInactive)
+                  case (Some(KeyActive(cid)), _) if cid == targetId =>
+                    keys.updated(gkey, KeyInactive)
+                  case (None, Some(KeyActive(cid))) if cid == targetId =>
+                    keys.updated(gkey, KeyInactive)
```
Comment (remyhaemmerle-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala:495]:
```suggestion
                (keys.get(gkey).orElse(globalKeyInputs.get(gkey)) match {
                  // An archive can only mark a key as inactive
                  // if it was brought into scope before.
                  case Some(KeyActive(cid)) if cid == targetId => keys.updated(gkey, KeyInactive)
}
  
```

**Review by remyhaemmerle-da — APPROVED** [review]
LGTM.

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala (hunk 2 above)
Comment (cocreature) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala]: 8443bdfc7f6baef1f5ce0c1b81598ccde0f06fb2

**Pull request merged**